### ISSUE TITLE
Functionnal utilities seem to not be fully handled

### ISSUE THREAD

**Issue by oltodo** [sergioazoc/oxlint-tailwindcss#104]
**Reproduction**

I get the following error when using a utility with a wildcard name:

```
tailwindcss(no-unknown-classes): "foo-1" is not a valid Tailwind class. Did you mean "top-1"?
```

Minimal example:

```css
@utility foo-* {
  --foo: --value(integer);
}
```

```tsx
function Component() {
  return <div className="foo-1" />;
}
```

**What I expected**

Classes generated by `@utility foo-*` (for example `foo-1`) should be recognized as valid Tailwind classes and not reported by the `no-unknown-classes` rule.

**Note**

Utilities with literal values or matching theme values look to be working properly:

```css
@theme {
  --bar-123: 123;
  --bar-456: 456;
  --bar-789: 789;
}

@utility bar-* {
  --bar: --value(--bar- *);
}

@utility baz-* {
  --baz: --value("a", "b", "c");
}
```

**Sandbox**

https://codesandbox.io/p/devbox/nhtzvw

--- 

Thanks for the great work on this plugin — much appreciated!

**Comment by sergioazoc** [comment]
Thanks @oltodo, and thanks especially for including the `bar-*` / `baz-*` contrast. That is what pointed straight at the mechanism.

`--value(integer)` declares an open value type, so there is nothing to enumerate. Tailwind's class list returns no entry for `foo-*` at all, not `foo-1`, not even the root `foo`. A value type the theme bounds (`--value(--bar-*)`) or a literal set (`--value("a","b","c")`) does enumerate, which is why those two worked and this one read as a typo of `top-1`.

The annoying part: the plugin was already asking the design system about `foo-1` on every lint, and it was already answering `.foo-1 { --foo: 1 }`. That answer was only wired up to reject a guess, never to confirm one, so the exact verdict got thrown away in favour of the Levenshtein neighbour.

Fixed in [1.7.1](https://www.npmjs.com/package/oxlint-tailwindcss/v/1.7.1), out now. The same change turned out to cover 3921 percentage classes that Tailwind compiles but does not enumerate (`from-33%`, `mask-b-from-70%`, `font-stretch-57%`), where the quick-fix was offering the nearest enumerated value, moving your gradient stop and leaving the lint green.

What still reports under `--value(integer)`: `foo-abc`, `foo-1.5` and `foo-[3]`, since Tailwind compiles none of them. Under a project prefix, a bare `foo-1` now asks for `tw:foo-1` instead of being called unknown.

Ping me if 1.7.1 doesn't clear it on your side.
